modules/decoding.py

```diff
@@ -0,0 +1,65 @@
+import chess
+from chess import Move, Board
+from chess.engine import Cp, Mate, BestMove
+from chess.uci import Score
+
+from modules.puzzle.analysed import analysed
+from modules.puzzle.position_list import position_list
+from modules.puzzle.puzzle import puzzle
+
+
+def score_from_dict(d) -> Score:
+    if not d:
+        return None
+    if d[0]:
+        return Cp(d[0])
+    else:
+        return Mate(d[1])
+    # return Score(d[0], d[1]) if d else None
```

Comment:
Probably delete this comment

Reply:
I’ll do that and rebase against the current master.